slsim/Sources/SourceTypes/extended_source.py

```diff
@@ -0,0 +1,160 @@
+from slsim.Sources.SourceTypes.single_sersic import SingleSersic
+from slsim.Sources.SourceTypes.double_sersic import DoubleSersic
+from slsim.Sources.SourceTypes.interpolated_image import Interpolated
+
+_SUPPORTED_EXTENDED_SOURCES = ["single_sersic", "double_sersic", "interpolated"]
+
+
+class ExtendedSource(object):
+    """Class to manage a single extended source."""
+
+    def __init__(self, source_dict, cosmo=None, **kwargs):
+        """
+        :param source_dict: Source properties. May be a dictionary or an Astropy table.
+         For a detailed description of this dictionary, please see the documentation for
+         the SingleSersic, DoubleSersic, and Interpolated classes.
+        :type source_dict: dict or astropy.table.Table
+        :param extendedsource_type: keyword for specifying light profile model.
+        :type extendedsource_type: str. supported types are "single_sersic",
+         "double_sersic", "interpolated".
+        :param cosmo: astropy.cosmology instance
+        :param kwargs: dictionary of keyword arguments for a extended source.
+         eg: kwargs = {"extendedsource_type": "single_sersic"}. Other supported
+         types are "single_sersic", "double_sersic", "interpolated".
+        """
+        self.extendedsource_type = kwargs["extendedsource_type"]
+        if self.extendedsource_type in ["single_sersic"]:
+            self._source = SingleSersic(source_dict=source_dict)
+        elif self.extendedsource_type in ["double_sersic"]:
+            self._source = DoubleSersic(source_dict=source_dict)
+        elif self.extendedsource_type in ["interpolated"]:
+            self._source = Interpolated(source_dict=source_dict, cosmo=cosmo)
+        else:
+            raise ValueError(
+                "Extended source type %s not supported. Chose among %s."
+                % (self.extendedsource_type, _SUPPORTED_EXTENDED_SOURCES)
+            )
+
+    @property
+    def redshift(self):
+        """Returns source redshift."""
+
+        return self._source.redshift
+
+    @property
+    def angular_size(self):
+        """Returns angular size of the source."""
+
+        return self._source.angular_size
+
+    @property
+    def ellipticity(self):
+        """Returns ellipticity components of source.
+        Defined as:
+
+        .. math::
+            e1 = \\frac{1-q}{1+q} * cos(2 \\phi)
+            e2 = \\frac{1-q}{1+q} * sin(2 \\phi)
+
+        with q being the minor-to-major axis ratio.
+        """
+
+        return self._source.ellipticity
+
+    @property
+    def n_sersic(self):
+        """Returns sersic indices of the source profile."""
+
+        return self._source.n_sersic
+
+    @property
+    def sersicweight(self):
+        """Returns weight of the sersic components."""
+
+        return self._source.sersicweight
+
+    @property
+    def image_redshift(self):
```

Comment:
I don't think you need access to this very specific thing of one particular source type. Find a way you can retrieve the information you need from the source without creating this access level